flink-connector-aws/flink-connector-aws-kinesis-streams/src/test/java/org/apache/flink/connector/kinesis/source/enumerator/tracker/SplitTrackerTest.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kinesis.source.enumerator.tracker;
+
+import org.apache.flink.connector.kinesis.source.enumerator.KinesisShardSplitWithAssignmentStatus;
+import org.apache.flink.connector.kinesis.source.enumerator.SplitAssignmentStatus;
+import org.apache.flink.connector.kinesis.source.split.KinesisShardSplit;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.apache.flink.connector.kinesis.source.util.TestUtil.generateShardId;
+import static org.apache.flink.connector.kinesis.source.util.TestUtil.getTestSplit;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+class SplitTrackerTest {
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testSplitWithoutParentAreAvailableToAssign(boolean preserveShardOrdering) {
+        SplitTracker splitTracker = new SplitTracker(preserveShardOrdering);
+
+        KinesisShardSplit split = getTestSplit(generateShardId(1), Collections.emptySet());
+
+        splitTracker.addSplits(Collections.singletonList(split));
+
+        List<KinesisShardSplit> pendingSplits = splitTracker.splitsAvailableForAssignment();
+
+        assertThat(pendingSplits).containsExactly(split);
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testStateSnapshot(boolean preserveShardOrdering) {
```

Comment:
nit: Might help readability to add some comments around the test structure

`// Given mixture of assigned and unassigned splits`

`// When registered with split tracker`

`// Then snapshot state accurately reflects assigned and unassigned splits`